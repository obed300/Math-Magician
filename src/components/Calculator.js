import React, { useState } from 'react';
import calculate from './Logic/Calc';

function Calculator() {
  const [calculation, setCalculation] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    const buttonName = e.currentTarget.textContent;
    const newState = calculate(calculation, buttonName);
    setCalculation(newState);
  };

  const { total, next, operation } = calculation;

  return (
    <div className="main-wrapper">
      <h3>Let us do some math!</h3>
      <div className="calContainer">
        <div className="input">
          {total}
          {operation}
          {next}
        </div>
        <ul className="second">
          <li><button name="AC" type="button" onClick={handleClick}>AC</button></li>
          <li><button name="+/-" type="button" onClick={handleClick}>+/-</button></li>
          <li><button name="%" type="button" onClick={handleClick}>%</button></li>
          <li>
            <button name="รท" className="yellow" type="button" onClick={handleClick}>
              รท
            </button>
          </li>
        </ul>

        <ul className="second">
          <li><button name="7" type="button" onClick={handleClick}>7</button></li>
          <li><button name="8" type="button" onClick={handleClick}>8</button></li>
          <li><button name="9" type="button" onClick={handleClick}>9</button></li>
          <li>
            <button name="x" className="yellow" type="button" onClick={handleClick}>
              x
            </button>
          </li>
        </ul>

        <ul className="second">
          <li><button name="4" type="button" onClick={handleClick}>4</button></li>
          <li><button name="5" type="button" onClick={handleClick}>5</button></li>
          <li><button name="6" type="button" onClick={handleClick}>6</button></li>
          <li>
            <button name="-" className="yellow" type="button" onClick={handleClick}>
              -
            </button>
          </li>
        </ul>

        <ul className="second">
          <li><button name="1" type="button" onClick={handleClick}>1</button></li>
          <li><button name="2" type="button" onClick={handleClick}>2</button></li>
          <li><button name="3" type="button" onClick={handleClick}>3</button></li>
          <li>
            <button name="+" className="yellow" type="button" onClick={handleClick}>
              +
            </button>
          </li>
        </ul>

        <ul className="second">
          <li className="zero">
            <button name="0" type="button" onClick={handleClick}>
              0
            </button>
          </li>
          <li><button name="." type="button" onClick={handleClick}>.</button></li>
          <li>
            <button name="=" className="yellow" type="button" onClick={handleClick}>
              =
            </button>
          </li>
        </ul>
      </div>

    </div>

  );
}

export default Calculator;
