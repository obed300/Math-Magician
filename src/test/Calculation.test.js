import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  it('should render correctly and match the saved snapshot', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
