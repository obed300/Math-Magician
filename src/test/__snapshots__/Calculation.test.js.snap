// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should render correctly and match the saved snapshot 1`] = `
<div
  className="main-wrapper"
>
  <h3>
    Let us do some math!
  </h3>
  <div
    className="calContainer"
  >
    <div
      className="input"
    />
    <ul
      className="second"
    >
      <li>
        <button
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
      </li>
      <li>
        <button
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
      </li>
      <li>
        <button
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
      </li>
      <li>
        <button
          className="yellow"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </li>
    </ul>
    <ul
      className="second"
    >
      <li>
        <button
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </li>
      <li>
        <button
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </li>
      <li>
        <button
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </li>
      <li>
        <button
          className="yellow"
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </li>
    </ul>
    <ul
      className="second"
    >
      <li>
        <button
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </li>
      <li>
        <button
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </li>
      <li>
        <button
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </li>
      <li>
        <button
          className="yellow"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </li>
    </ul>
    <ul
      className="second"
    >
      <li>
        <button
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </li>
      <li>
        <button
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </li>
      <li>
        <button
          className="yellow"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </li>
    </ul>
    <ul
      className="second"
    >
      <li
        className="zero"
      >
        <button
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </li>
      <li>
        <button
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </li>
      <li>
        <button
          className="yellow"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </li>
    </ul>
  </div>
</div>
`;
